artio-core/src/main/java/uk/co/real_logic/artio/engine/EngineContext.java

```diff
@@ -375,10 +375,16 @@ private void newArchivingAgent()
                 clock);
         }
 
+        final Agent dutyCycleTrackingAgent = new IndexerDutyCycleTracker(
+            configuration.agentNamePrefix(),
+            clock,
+            fixCounters.getIndexerDutyCycleTracker(configuration.indexerCycleThresholdNs()));
+
         final List<Agent> agents = new ArrayList<>();
         agents.add(inboundIndexer);
         agents.add(outboundIndexer);
         agents.add(replayer);
+        agents.add(dutyCycleTrackingAgent);
```

Comment:
While your solution is cleaner, I'd like to avoid changing thread names, can we squeeze that tracking into one of the existing agents?